samples/subsys/usb/webusb-next/src/msosv2.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2016-2019 Intel Corporation
+ * Copyright (c) 2023-2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MSOSV2_DESCRIPTOR_H
+#define ZEPHYR_INCLUDE_MSOSV2_DESCRIPTOR_H
+
+/*
+ * Microsoft OS 2.0 platform capability and Microsoft OS 2.0 descriptor set.
+ * See Microsoft OS 2.0 Descriptors Specification for reference.
+ */
+
+#define SAMPLE_MSOS2_VENDOR_CODE	0x02U
+/* Windows version (10)*/
+#define SAMPLE_MSOS2_OS_VERSION		0x0A000000UL
+
+/* random GUID {FA611CC3-7057-42EE-9D82-4919639562B3} */
+#define WEBUSB_DEVICE_INTERFACE_GUID \
+	'{', 0x00, 'F', 0x00, 'A', 0x00, '6', 0x00, '1', 0x00, '1', 0x00, \
+	'C', 0x00, 'C', 0x00, '3', 0x00, '-', 0x00, '7', 0x00, '0', 0x00, \
+	'5', 0x00, '7', 0x00, '-', 0x00, '4', 0x00, '2', 0x00, 'E', 0x00, \
+	'E', 0x00, '-', 0x00, '9', 0x00, 'D', 0x00, '8', 0x00, '2', 0x00, \
+	'-', 0x00, '4', 0x00, '9', 0x00, '1', 0x00, '9', 0x00, '6', 0x00, \
+	'3', 0x00, '9', 0x00, '5', 0x00, '6', 0x00, '2', 0x00, 'B', 0x00, \
+	'3', 0x00, '}', 0x00, 0x00, 0x00, 0x00, 0x00
+
+struct msosv2_descriptor {
+	struct msosv2_descriptor_set_header header;
```

Comment:
If not implement it right now, at least add a TODO: Add subset header when there is more than one interface.

See #66449 for more information.

Reply:
I have aligned it with the descriptor in the legacy sample and added a comment.